runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityPlugin.java

```diff
@@ -421,4 +427,56 @@ private void onTileObject(Tile tile, TileObject oldObject, TileObject newObject)
 			}
 		}
 	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (!config.showShortcutLevel())
+		{
+			return;
+		}
+
+		if (event.getTarget() == null || event.getTarget() == "")
+		{
+			return;
+		}
+
+		if (event.getType() == MenuAction.EXAMINE_OBJECT.getId()
```

Comment:
Is there a specific menu action type that's used for agility obstacles? `GAME_OBJECT_FIRST_OPTION` probably? It might be better to guard against that than to skip these ones specifically.

eg.

```java
if (event.getType() != MenuAction.GAME_OBJECT_FIRST_OPTION.getId()
    && event.getType() != MenuAction.GAME_OBJECT_SECOND_OPTION.getId())
    // continue game object menu action checks here as needed
```